### PR TITLE
Add description for initializer dtype

### PR DESCRIPTION
**Pull request by Crissman**
Addressing #4897

### CODE REVIEW

**Comment by Crissman** [comment]
Travis error looks unrelated.

**Comment by niboshi** [comment]
What are "`initializers`" and "`initializer`", respectively?

I think there is no such thing as "dtype" argument in initializers.

Also, this describes how to use dtype from the caller's perspective. Instead, we should describe the requirements of an initiaizer. In other words we should mention the intrinsic `dtype` attribute of initializer objects.

**Comment by Crissman** [comment]
@niboshi Edited accordingly. If this isn't what you're looking for, please provide example text for what you'd like it to say.

Thanks.

**Comment by niboshi** [comment]
How about this? This is what I suggested previously (with some edit).

>If an initializer object has `dtype` attribute, the initializer can assume that the array to feed the data into has that dtype. If the required dtype, depending on the context where the initializer is used, does not match the `dtype` attribute, Chainer will report an error.

**Comment by niboshi** [comment]
"Base class" section is broken. Please fix it.
Also, some words have to be marked up correctly as identifiers: "`dtype` attribute" (there are two instances).

**Comment by Crissman** [comment]
Admended. Please confirm. @niboshi

File: docs/source/reference/initializers.rst [hunk 1]
```diff
@@ -19,6 +19,8 @@ A weight initializer can be any of the following objects.
   Unless explicitly specified, it is :class:`~chainer.initializers.LeCunNormal`
   with scale value 1.
 
+If an `initializer` object has the `~numpy.dtype` attribute, the `initializer` can assume that the array to feed the data into has that `~numpy.dtype`. If the required dtype, depending on the context where the `initializer` is used, does not match the `~numpy.dtype` attribute, Chainer will report an error.
```
[the remark this answers is not in the document]
Comment (niboshi) [reply docs/source/reference/initializers.rst]: Please fix the markups.
* "initializer" is not an identifier, but a general noun here.
* "dtype" is also a general noun. Only those refer to the attribute are identifiers.

**Comment by Crissman** [comment]
@niboshi Thanks! Done.

**Comment by niboshi** [comment]
jenkins, test this please

File: docs/source/reference/initializers.rst [hunk 2]
```diff
@@ -19,6 +19,8 @@ A weight initializer can be any of the following objects.
   Unless explicitly specified, it is :class:`~chainer.initializers.LeCunNormal`
   with scale value 1.
 
+If an initializer object has the `dtype` attribute, the initializer can assume that the array to feed the data into has that dtype. If the required dtype, depending on the context where the initializer is used, does not match the `dtype` attribute, Chainer will report an error.
```
Comment (niboshi) [inline docs/source/reference/initializers.rst]: Oops, we need double graves.

```suggestion
If an initializer object has the ``dtype`` attribute, the initializer can assume that the array to feed the data into has that dtype. If the required dtype, depending on the context where the initializer is used, does not match the ``dtype`` attribute, Chainer will report an error.
```

**Comment by Crissman** [comment]
Ok, now with double graves! Please merge, @niboshi .

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**